test/e2e/tests/multichain/asset-list.spec.ts

```diff
@@ -0,0 +1,192 @@
+        const assetListPage = new AssetListPage(driver);
+        await headerNavbar.clickSwitchNetworkDropDown();
+        await selectNetworkDialog.selectNetworkName(NETWORK_NAME_MAINNET);
+        await assetListPage.waitUntilAssetListHasItems(2);
+        await driver.clickElement('.multichain-token-list-item');
+        const coinOverviewElement = await driver.findElement(
+          '[data-testid="coin-overview-buy"]',
+        );
+        const multichainTokenListButton = await driver.findElement(
+          '[data-testid="multichain-token-list-button"]',
+        );
+        assert.ok(coinOverviewElement, 'coin-overview-buy is present');
+        assert.ok(
+          multichainTokenListButton,
+          'multichain-token-list-button is present',
+        );
+      },
+    );
+  });
+  it('switches networks when clicking on send for a token on another network', async function () {
+    await withFixtures(
+      buildFixtures(this.test?.fullTitle() as string),
+      async ({
+        driver,
+        ganacheServer,
+      }: {
+        driver: Driver;
+        ganacheServer?: Ganache;
+      }) => {
+        await loginWithBalanceValidation(driver, ganacheServer);
+        const assetListPage = new AssetListPage(driver);
+        const sendPage = new SendTokenPage(driver);
+        await assetListPage.waitUntilAssetListHasItems(4);
+        await assetListPage.clickOnAsset('LineaETH');
+        await driver.clickElement('[data-testid="coin-overview-send"]');
+        await sendPage.check_networkChange('Linea Sepolia');
+        await sendPage.check_pageIsLoaded();
+        await sendPage.fillRecipient(
+          '0x2f318C334780961FB129D2a6c30D0763d9a5C970',
+        );
+        await sendPage.clickAssetPickerButton();
+        const assetPickerItems = await sendPage.getAssetPickerItems();
+        assert.equal(
+          assetPickerItems.length,
+          1,
+          'Only one asset should be shown in the asset picker',
+        );
+      },
+    );
+  });
+  it('switches networks when clicking on swap for a token on another network', async function () {
+    await withFixtures(
+      buildFixtures(this.test?.fullTitle() as string),
+      async ({
+        driver,
+        ganacheServer,
+      }: {
+        driver: Driver;
+        ganacheServer?: Ganache;
+      }) => {
+        await loginWithBalanceValidation(driver, ganacheServer);
+        const assetListPage = new AssetListPage(driver);
+        await assetListPage.waitUntilAssetListHasItems(4);
+        await assetListPage.clickOnAsset('LineaETH');
+        await driver.clickElement('[data-testid="token-overview-button-swap"]');
+        const toastTextElement = await driver.findElement('.toast-text');
+        const toastText = await toastTextElement.getText();
+        assert.equal(
+          toastText,
+          "You're now using Linea Sepolia",
+          'Toast text is correct',
+        );
+      },
+    );
+  });
```

Comment:
If we do end up doing a try/catch, or some sort of error handling on the client to prevent the "network switch failure" case, I'd imagine that we'd probably add some additional e2e tests here as well.